cmd/find.go

```diff
@@ -0,0 +1,472 @@
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		for {
+			select {
+			case <-trapCh:
+				console.Println()
+				close(watchObj.doneChan)
+				return
+			case event, ok := <-watchObj.Events():
+				if !ok {
+					return
+				}
+				msg := contentMessage{}
+				msg.Key = strings.TrimPrefix(event.Path, cliTempURL)
+				msg.Time, _ = time.Parse(time.RFC822, event.Time)
+				msg.Size = event.Size
+
+				msg.Key = alias + msg.Key
+
+				//check to see if the newly create object matches the given params
+				match := isMatch(msg, ctx.Bool("or"), ctx)
+
+				if match && ctx.String("exec") != "" {
+					execFlag(ctx, msg.Key, fileContent)
+				}
+
+			case err, ok := <-watchObj.Errors():
+				if !ok {
+					return
+				}
+				errorIf(err, "Unable to watch for events.")
+				return
+			}
+		}
+	}()
+
+	wg.Wait()
+}
+
+func isMatch(fileContent contentMessage, orOp bool, ctx *cli.Context) bool {
+	match := true
+
+	if (match && !orOp) && ctx.String("ignore") != "" {
+		match = !pathFlag(fileContent.Key, ctx.String("ignore"))
+	}
+
+	//verify that the newly added object matches all of the other specified params
+	if ctx.String("name") != "" {
+		temp1, err1 := nameFlag(fileContent.Key, ctx.String("name"))
+		match = temp1
+
+		if err1 != nil {
+			console.Println("Name could not be matched")
+			console.Fatalln()
+		}
+	}
+
+	if (match || orOp) && ctx.String("path") != "" {
+		match = pathFlag(fileContent.Key, ctx.String("path"))
+	}
+
+	if (match || orOp) && ctx.String("regex") != "" {
+		temp, e := regexFlag(fileContent.Key, ctx.String("regex"))
+		match = temp
+
+		if e != nil {
+			console.Println("Regex could not be matched")
+			console.Fatalln()
+		}
+	}
+
+	if (match || orOp) && ctx.String("older") != "" {
+		match = olderFlag(fileContent.Time, ctx.String("older"))
+	}
+
+	if (match || orOp) && ctx.String("newer") != "" {
+		match = newerFlag(fileContent.Time, ctx.String("newer"))
+	}
+
+	if (match || orOp) && ctx.String("larger") != "" {
+		match = largerFlag(fileContent.Size, ctx.String("larger"))
+	}
+
+	if (match || orOp) && ctx.String("smaller") != "" {
+		match = smallerFlag(fileContent.Size, ctx.String("smaller"))
+	}
+
+	return match
+}
+
+func timeHelper(pattern string) (int, error) {
+	conversion := map[string]int{
+		"dy": 1,
+		"wk": 7,
+		"mh": 30,
+		"yr": 365,
+	}
+	if int1, err1 := strconv.Atoi(pattern); err1 == nil {
+		return int1, nil
+	}
+
+	t := pattern[len(pattern)-2:]
+	i, err := strconv.Atoi(pattern[:len(pattern)-2])
+
+	if err != nil {
+		return 0, err
+	}
+	return i * conversion[strings.ToLower(t)], nil
+}
+
+func getPurl(filepath string) string {
+	targetAlias, targetURLFull, _, err := expandAlias(filepath)
+	if err != nil {
+		console.Println("Error with expand alias")
+		console.Fatalln()
+	}
+	clnt, err := newClientFromAlias(targetAlias, targetURLFull)
+	if err != nil {
+		console.Println("Error with newClientFromAlias")
+		console.Fatalln()
+	}
+
+	shareDB := newShareDBV1()
```

Comment:
Don't need to generate an updated the shareDB file IMO, since find executes and completes quickly as it processes. There is no reason to keep inside the DB.  Just do `client.ShareDownload()`